### PR TITLE
Api class refactor

### PR DESCRIPTION
**Pull request by piaskowyk**

### CODE REVIEW

File: src/maintenance/Guardian.cpp [hunk 1]
```diff
@@ -34,7 +33,7 @@ String Guardian::getDeviceState() {
   #ifndef IS_MOCK
   state += ",";
   state += "API_CONNECTION|";
-  state += core.isError() ? "ERROR" : "OK";
+  state += "OK"; // temporary
```
Comment (jakubdybczak) [inline src/maintenance/Guardian.cpp:36]: temporary

File: backtrace.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import os
```
Comment (jakubdybczak) [inline backtrace.py:1]: delete this file

File: schema [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// REGISTRATION
```
Comment (jakubdybczak) [inline schema:1]: Also to be deleted

File: include/api/Api.h [hunk 4]
```diff
@@ -2,12 +2,13 @@
...
-#include "util/DataPersister.h"
 #include "communication/common/Internet.h"
 #include "config/Config.h"
 #include "maintenance/Logger.h"
 #include "communication/bluetooth/Bluetooth.h"
 #include "maintenance/Statistics.h"
+#include "api/entity/RegiserModel.h"
+#include "api/entity/DataModel.h"
 
 #define ACCESS_TOKEN_EXPIRATION_TIME 600000  // 1000ms * 60 * 10 = 10 min
```
Comment (jakubdybczak) [inline include/api/Api.h:13]: Maybe should we change this to 24 hours

**Review by jakubdybczak — CHANGES_REQUESTED** [review]

File: include/communication/gsm/GsmConn.h [hunk 5]
```diff
@@ -10,14 +10,18 @@ class GsmConn {
...
+  static Http::Response httpGet(const String& url, const String& authorization);
+  static Http::Response httpPost(const String& url, const String body, const String& authorization);
+  static Http::Response httpPut(const String& url, const String body, const String& authorization);
   static bool isConnected();
   static bool isOk();
   static GSM gsm;
 
  private:
-  static Http::Response buildRequest(const String& url, Http::Method method, String& authorization,
-                                     String* body = NULL);
```
Comment (matix522) [inline include/communication/gsm/GsmConn.h:22]: use nullptr

**Review by matix522 — CHANGES_REQUESTED** [review]

**Pull request merged**